src/language/patterns.md

```diff
@@ -0,0 +1,362 @@
+---
+title: Patterns
+description: Summary of patterns in Dart.
+---
+
+{{site.alert.version-note}}
+  Patterns require a [language version][] of at least 3.0.
+{{site.alert.end}}
+
+Patterns are a syntactic category in the Dart language, like statements and expressions.
+A pattern represents the shape of a set of values that it may match against actual
+values.
+
+This page describes:
+- What patterns do.
+- Where patterns are allowed in Dart code.
+- What the common use cases for patterns are.
+
+To learn about the different kinds of patterns, visit the [pattern types][types]
+page.
+
+## What patterns do
+
+In general, a pattern may **match** a value, **destructure** a value, or both,
+depending on the context and shape of the pattern.
+
+First, _pattern matching_ allows you to check whether a given value:
+- Has a certain shape.
+- Is a certain constant.
+- Is equal to something else.
+- Has a certain type.
+
+Then, _pattern destructuring_ provides you with a convenient declarative syntax to
+break that value into its constituent parts. The same pattern can also let you 
+bind variables to some or all of those parts in the process.
+
+### Matching
+
+A pattern always tests against a value to determine if the value has the form
+you expect. In other words, you are checking if the value _matches_ the pattern. 
+
+What constitutes a match depends on [what kind of pattern][types] you are using.
+For example, a constant pattern matches if the value is equal to the pattern's 
+constant:
+
+```dart
+switch (number) {
+  // Constant pattern matches if 1 == number.
+  case 1: // ...        
+}
+```
+
+Many patterns make use of subpatterns, sometimes called _outer_ and _inner_
+patterns, respectively. Patterns match recursively on their subpatterns.
+For example, the individual fields of any [collection-type][] pattern could be 
+[variable patterns][variable] or [constant patterns][constant]:
+
+```dart
+switch (obj) {
+  // List pattern [a,b] matches obj first if obj is a list wth two fields,
```

Comment:
Missing `i`

```suggestion
  // List pattern [a, b] matches obj first if obj is a list with two fields,
```

Reply:
Thanks for catching this! Could you open a separate PR targeting this branch with a fix? Or I can. We just don't want to commit directly to this tracking branch.